plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/TestClickHouseConnectorTest.java

```diff
@@ -709,24 +709,11 @@ protected TestTable simpleTable()
         return new TestTable(onRemoteDatabase(), "tpch.simple_table", "(col BIGINT) Engine=Log", ImmutableList.of("1", "2"));
     }
 
-    @Test
     @Override
-    public void testCreateTableWithLongTableName()
+    protected void verifyTableNameLengthFailurePermissible(Throwable e)
     {
-        // Override because ClickHouse connector can create a table which can't be dropped
-        String baseTableName = "test_create_" + randomNameSuffix();
-        String validTableName = baseTableName + "z".repeat(maxTableNameLength().orElseThrow() - baseTableName.length());
-
-        assertUpdate("CREATE TABLE " + validTableName + " (a bigint)");
-        assertThat(getQueryRunner().tableExists(getSession(), validTableName)).isTrue();
-        assertThatThrownBy(() -> assertUpdate("DROP TABLE " + validTableName))
-                .hasMessageMatching("(?s).*(Bad path syntax|File name too long).*");
-
-        String invalidTableName = baseTableName + "z".repeat(maxTableNameLength().orElseThrow() - baseTableName.length() + 1);
-        assertThat(query("CREATE TABLE " + invalidTableName + " (a bigint)"))
-                .failure().hasMessageMatching("(?s).*(Cannot open file|File name too long).*");
-        // ClickHouse lefts a table even if the above statement failed
-        assertThat(getQueryRunner().tableExists(getSession(), validTableName)).isTrue();
+        assertThat(e).hasMessageContaining("The max length of table name for database tpch is %d, current length is %d"
+                .formatted(maxTableNameLength().orElseThrow(), maxTableNameLength().orElseThrow() + 1));
```

Comment:
Not sure if I fully understand. 
Here I'm addressing https://github.com/trinodb/trino/pull/24515#discussion_r1895664288
Also this message is generated by Clickhouse.

Reply:
https://github.com/trinodb/trino/pull/24515#discussion_r1895664288 here we are applying a full regex where we  match both the limitation by Clickhouse and the tableName we create. I think we should have a strict check for the Clickhouse limitation i.e `The max length of table name for database tpch is %d` while the other part of the error message can be a regex-pattern as it could change independent of the Clickhouse version.